instant-xml/src/impls.rs

```diff
@@ -25,3 +28,149 @@ impl<'xml> FromXml<'xml> for bool {
+                        serializer.add_attribute_value(value);
+                    }
+                    _ => {
+                        serializer.add_open_tag(&field_context)?;
+                        serializer.output.write_str(value)?;
+                        serializer.add_close_tag(field_context)?;
+                    }
+                }
+                Ok(())
+            }
+            None => Err(Error::UnexpectedValue),
+        }
+    }
+}
+
+impl ToXml for String {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        match serializer.consume_field_context() {
+            Some(field_context) => {
+                match field_context.attribute {
+                    Some(FieldAttribute::Attribute) => {
+                        serializer.add_attribute_value(self);
+                    }
+                    _ => {
+                        serializer.add_open_tag(&field_context)?;
+                        serializer.output.write_str(self)?;
+                        serializer.add_close_tag(field_context)?;
+                    }
+                }
+                Ok(())
+            }
+            None => Err(Error::UnexpectedValue),
+        }
+    }
+}
+
+impl ToXml for char {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        match serializer.consume_field_context() {
+            Some(field_context) => {
+                let mut tmp = [0u8; 4];
+                let char_str = self.encode_utf8(&mut tmp);
+                match field_context.attribute {
+                    Some(FieldAttribute::Attribute) => {
+                        serializer.add_attribute_value(char_str);
+                    }
+                    _ => {
+                        serializer.add_open_tag(&field_context)?;
+                        serializer.output.write_str(char_str)?;
+                        serializer.add_close_tag(field_context)?;
+                    }
+                }
+                Ok(())
+            }
+            None => Err(Error::UnexpectedValue),
+        }
+    }
+}
+
+impl ToXml for &str {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        match serializer.consume_field_context() {
+            Some(field_context) => {
+                match field_context.attribute {
+                    Some(FieldAttribute::Attribute) => {
+                        serializer.add_attribute_value(self);
+                    }
+                    _ => {
+                        serializer.add_open_tag(&field_context)?;
+                        serializer.output.write_str(self)?;
+                        serializer.add_close_tag(field_context)?;
+                    }
+                }
+                Ok(())
+            }
+            None => Err(Error::UnexpectedValue),
+        }
+    }
+}
```

Comment:
Please also add an impl for `Cow<'_, str>`, which we will need soon.

Reply:
done